daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -150,12 +149,14 @@ func New(info logger.Info) (logger.Logger, error) {
 		return nil, err
 	}
 
+	logNonBlocking := info.Config["mode"] == "non-blocking"
+
 	containerStream := &logStream{
 		logStreamName:      containerStreamConfig.logStreamName,
 		logGroupName:       containerStreamConfig.logGroupName,
 		logCreateGroup:     containerStreamConfig.logCreateGroup,
 		logCreateStream:    containerStreamConfig.logCreateStream,
-		logNonBlocking:     containerStreamConfig.logNonBlocking,
+		logNonBlocking:     logNonBlocking,
```

Comment:
The only place this field is read is inside this very function, eight lines down. Change line 167 to `if logNonBlocking {` and the `logStream.logNonBlocking` field can also go away.

Reply:
Good call. It is silly that I didn't realize that. Thanks/updated!